### PR TITLE
Long slugs breaking summary panel UI

### LINKED ISSUE #64052: Long slugs break Summary panel UI
Screenshot 2024-07-29 at 11 07 29

### PR DESCRIPTION
**Pull request by amitraj2203**
## What?
Fixes: #64052 

## Why?
See this [comment](https://github.com/WordPress/gutenberg/issues/64052#issue-2435002740)

## Screenshots or screencast 
image

### CODE REVIEW

**Comment by jameskoster** [comment]
Thank you for the PR 🙏 

I'm not sure this is the correct solution though, it's making the other buttons in the panel `36px` tall rather than the intended `32px`:

Screenshot 2024-07-29 at 14 35 03

Another option to try could be; target `.editor-post-url__panel-toggle` and apply `height: auto`, what do you think?

**Comment by jameskoster** [comment]
Alternatively it might be better to truncate the button. When the URL is _extremely_ long this button can consume a large amount of space. 

Screenshot 2024-07-29 at 14 41 38

**Comment by amitraj2203** [comment]
@jameskoster Thanks for the review. I have changed the CSS to truncate URL when they are extremely long.

**Comment by jameskoster** [comment]
Thank you! This is working for me, let's get a review from the components team.

File: packages/editor/src/components/post-url/style.scss [hunk 1]
```diff
@@ -27,6 +27,12 @@
 	padding-inline-start: 0 !important;
 }
 
-.editor-post-url__panel-toggle {
-	word-break: break-word;
+.editor-post-panel__row-control {
+	.components-button.editor-post-url__panel-toggle {
```
Comment (ciampo) [inline packages/editor/src/components/post-url/style.scss]: Let's avoid using `.components-*` classnames, as they are not assumed to be public APIs from the `@wordpress/components` package.

**Review by ciampo — CHANGES_REQUESTED** [review]
Thank you for the ping, @jameskoster !

We do have a [`Truncate` component](https://wordpress.github.io/gutenberg/?path=/docs/components-experimental-truncate--docs) that should this use case perfectly — the only tweak that we'd have to make for the best results is to make sure that `children` of `<Truncate />` are strings (and not JSX).

```diff
diff --git a/packages/editor/src/components/post-url/panel.js b/packages/editor/src/components/post-url/panel.js
index be32b40eaf..aca36566c0 100644
--- a/packages/editor/src/components/post-url/panel.js
+++ b/packages/editor/src/components/post-url/panel.js
@@ -3,7 +3,11 @@
  */
 import { useMemo, useState } from '@wordpress/element';
 import { useSelect } from '@wordpress/data';
-import { Dropdown, Button } from '@wordpress/components';
+import {
+	Dropdown,
+	Button,
+	__experimentalTruncate as Truncate,
+} from '@wordpress/components';
 import { __, sprintf } from '@wordpress/i18n';
 import { safeDecodeURIComponent } from '@wordpress/url';
 import { store as coreStore } from '@wordpress/core-data';
@@ -86,7 +90,9 @@ function PostURLToggle( { isOpen, onClick } ) {
 			aria-label={ sprintf( __( 'Change link: %s' ), decodedSlug ) }
 			onClick={ onClick }
 		>
-			{ isFrontPage ? postLink : <>/{ decodedSlug }</> }
+			<Truncate numberOfLines={ 1 }>
+				{ isFrontPage ? postLink : `/${ decodedSlug }` }
+			</Truncate>
 		</Button>
 	);
 }
diff --git a/packages/editor/src/components/post-url/style.scss b/packages/editor/src/components/post-url/style.scss
index 45e5b2063d..104f246458 100644
--- a/packages/editor/src/components/post-url/style.scss
+++ b/packages/editor/src/components/post-url/style.scss
@@ -27,12 +27,6 @@
 	padding-inline-start: 0 !important;
 }
 
-.editor-post-panel__row-control {
-	.components-button.editor-post-url__panel-toggle {
-		word-break: break-word;
-		display: inline-block;
-		text-overflow: ellipsis;
-		overflow: hidden;
-		white-space: nowrap;
-	}
+editor-post-url__panel-toggle {
+	word-break: break-word;
 }

```

This shouldn't require any additional style overrides.

File: packages/editor/src/components/post-url/style.scss (hunk 1 above)
Comment (amitraj2203) [reply packages/editor/src/components/post-url/style.scss]: @ciampo Thanks for the review. Can you please review it again?

**Review by ciampo — APPROVED** [review]
Looking good, thank you! 🙏

**Review by jameskoster — APPROVED** [review]

**Comment by afercia** [comment]
I kindly disagree with this change. The entire purpose of this UI is ot show the _full_ link or slug. Truncation is not a content strategy. If there's not enough space in the design to show content, the root problem is in the design in the first place.

**Pull request merged**